terms/bls12-381.md

```diff
@@ -0,0 +1,325 @@
+With this $\chi$ value we have:
+
+| Parameter     |        | Equation                            | Value              | Comments        |
+| ------------- | ------ | ----------------------------------- | ------------------ | --------------- |
+| Field modulus | $q$    | $\frac{1}{3}(x-1)^2(x^4-x^2+1) + x$ | 0x1a0...aaab       | 381 bits, prime |
+| Subgroup size | $r$    | $(x^4 - x^2 + 1)$                   | 0x73e...000001     | 255 bits, prime |
+|               | $a$    |                                     | 0x00               |                 |
+|               | $b$    |                                     | 0x04               |                 |
+|               | $\chi$ |                                     | 0xd201000000010000 |                 |
+
+### Field Extensions
+
+The "12" in BLS12-381 is not only the embedding degree, it is also the degree of [extension fields](extension_fields.md).
+Let's construct $F_{q^2}$, the quadratic extension of $F_q$. Let's try this multiplication:
+$$(a, b) \times (c, d) = (a + bx)(c + dx)$$
+We need rules to reducing polynomials so that they
+have a degree less than two.
+
+1. It must be a degree $k$ polynomial, where $k$ is out extension degree, 2 in this case.
+2. It must be irreducible in the field we are extending. That means it must not be possible
+ to factor it into two or more lower degree polynomials.
+Applying our rule, by substituting $x^2 = -1$, gives us:
+$$(ac - bd) + (ad + bc)x = (ac - bd, ad + bc)$$
+For finite fields, if we can find an irreducible
+$k$-degree polynomial in our field $F_q$, then we are able
+to extend the field to $\mathbb{F_{q^{k}}}$, and represent the elements of the extended field as degree $k - 1$
+polynomials
+$$a_0 + a_1x + ... + a_{k-1}x^{k-1}$$
+We can represent this compactly as: $$(a_0, ..., a_{k-1})$$
+We can use [Twists](#twists) or [Extension Towers](terms/bls12-381.md#Extension%20Towers)) to
+reduce modular.
+
+### The Curves
+
+Actually, the BLS12-381 has two curves:
+
++ The simpler one is $E(F_q)$ over the [finite field](prime_or_finite_fields.md) $F_q$
++ The other curve $E'(F_{q^2})$ is defined over an [extension fields](extension_fields.md) of $F_q$ to $F_{q^2}$ (think complex number)
+
+### The Subgroups
+
+You should read [pairings or bilinear map](pairings_or_bilinear_map.md) before this section.
+
+If we keep extending the field over which $E$ is defined, it can be proved that we eventually
+find a curve that has more than one subgroup of order $r$ (in fact, $r$ + 1 of them).
+That is, for some $k$, $E'(F_{q^k})$ contains other subgroups of order $r$ that we can use.
+One of these subgroups contains only points having a trace of zero, and we choose that subgroups to be $G_2$.
+
+Since both $G_1$ in $E(F_q)$ and $G_2$ in $E(F_{q^{12}})$ have the same order $r$, we can now use them to do pairings.
+
+### Twists
+
+Twist can be used to transform $E(F_{q^{12}})$ curve into a curve defined over a lower degree field
```

Comment:
For efficiency reasons, Twist can be used to transform the $E(F_{q^{12}})$ curve into a curve defined over a lower degree field that still has an order $r$ subgroup.